pandas_gbq/exceptions.py

```diff
@@ -28,6 +28,10 @@ class InvalidPrivateKeyFormat(ValueError):
     """
 
 
+class LargeResultsWarning(UserWarning):
+    """Raise when results are beyond that recommended for pandas DataFrame."""
+
+
```

Comment:
nit - is there a threshold when this should be used? or is it up to our discretion? If specific guidance/threshold please add!

Reply:
Up to our discretion. We estimate this will be reached in maybe 5% of pandas-gbq usage, so hopefully won't be too annoying and 1 GB is quite large for a pandas DataFrame, but some posts say 100 GB is the hard limit for a pandas DataFrame (I haven't found an authoritative source for that, though).